test/xds/xds_server_rbac_test.go

```diff
@@ -843,8 +847,51 @@ func serverListenerWithBadRouteConfiguration(t *testing.T, host string, port uin
 func (s) TestRBAC_WithBadRouteConfiguration(t *testing.T) {
 	managementServer, nodeID, bootstrapContents, xdsResolver := setup.ManagementServerAndResolver(t)
 
-	lis, cleanup2 := setupGRPCServer(t, bootstrapContents)
-	defer cleanup2()
+	// Initialize a test gRPC server, assign it to the stub server, and start
+	// the test service.
+	lis, err := testutils.LocalTCPListener()
```

Comment:
I don't know how many tests that use an xDS-enabled gRPC server have this boilerplate that creates a listener, a stub server, a gRPC server, and then starts serving. The `stubserver.StartTestService` is very convenient for tests that don't need an xDS-enabled gRPC server. Maybe we can have a helper for the tests that need an xDS-enabled gRPC server, similar to `stubserver.StartTestService`? (and maybe we can get some TVC to work on that?).

Reply:
`setupGRPCServer` in this file is a helper that creates an xDS enabled gRPC server. I changed the test to to not use it because I wanted to set a server option for setting the serving mode callback. The helper was already setting this server option. I made the `setupGRPCServer` accept a variadic list of server options, similar to `StartTestService`. Users can override the default options set by `setupGRPCServer` as `setupGRPCServer` appends the passed options to the end of the list used to create the new server.